synapse/storage/databases/main/devices.py

```diff
@@ -222,13 +223,18 @@ async def get_device_updates_by_remote(
             limit,
         )
 
-        # We need to ensure `updates` doesn't grow too big.
-        # Currently: `len(updates) <= limit`.
+        # Note for later that `len(updates) <= limit`.
```

Comment:
Not sure if this comment change makes sense?

Reply:
Ahhh: when I first reviewed this I assumed we were growing `updates` in the rest of the function. We're not though: we gradually build up `results` instead. cc @reivilibre

Reply:
It might be an OK change, it just surprised me a bit to be in a PR about adding debug to figure out what is happening... 😄

Reply:
That's fair. I can pull that out for now.